lua/terrortown/menus/score/info.lua

```diff
@@ -49,20 +49,42 @@ local function MakePlayerScoreTooltip(parent, width, ply)
 	titleBox:SetTitle("tooltip_score_gained")
 	titleBox:SetTitleAlign(TEXT_ALIGN_LEFT)
 
+	-- In a first Pass filter all scoreevents and scores by name, positivy and number of Events
+	local filteredPlyScores = {}
 	for i = 1, #plyScores do
 		local plyScoreEvent = plyScores[i]
 		local rawScoreTexts = plyScoreEvent:GetRawScoreText(ply.sid64)
 
 		for k = 1, #rawScoreTexts do
 			local rawScoreText = rawScoreTexts[k]
+			local scoreName = rawScoreText.name
+			local score = rawScoreText.score
+
+			filteredPlyScores[scoreName] = filteredPlyScores[scoreName] or {score = {0,0}, numEvents = {0,0}}
+
+			local posIndex = score < 0 and 1 or 2 -- first entry contains negative and second positive entries
+			filteredPlyScores[scoreName].score[posIndex] = filteredPlyScores[scoreName].score[posIndex] + score
```

Comment:
empty line before this one

Reply:
And localize `filteredPlyScores[scoreName]`

Reply:
I thought of localizing ot, but what of its nil? Is it then still an object, that you can handle? I was afraid by localizing that, I might instantiate a new object, that I have to assign later. (Still not confident in use with lua....)

Reply:
First if all, it's not an object but a table :D
Secondly, it can't be nil, because you already integrated a fallback 3 lines above. if you localizing that result, it shouldn't fail.

Reply:
Okay like that it will actually save 3 of 4 calls. Readadbility only gets increased if I find a good name. Maybe change the first into ...List and then localize to filteredPlyScore. I would have loved to localize it in the first assignment, thatwould have fitted nicely into the flow. Like this it feels a bit unnecessary, but it kinda makes sense :D